docs/DeveloperGuide.md

```diff
@@ -1,38 +1,361 @@
+instance,
+and a base currency of USD, and our API token from Open Exchange Rates.
+
+getLatestExchangeRates returns a Call object, and we enqueue a Callback object to get the onResponse() and onFailure()
+methods
+that will be called depending on the outcome of the Call. If the call is successful, onResponse() returns an
+ExchangeRatesResponse object containing the HashMap of ISO currency tags as Strings for keys, and doubles for rates.
+This data
+is then extracted using saveMap, which filters out the rates for our supported currencies and performs type conversion.
+The savedMap attribute of ExchangeRates is set to this filtered map, which is then passed to Forex via getExchangeRates.
+
+If onFailure() is called, it means an unexpected error was encountered, such as losing Internet connection.
+
+Below is a UML Diagram of the classes and their respective methods.
+![APIClassDiagram](images/APIClassDiagram.png)
+
+### Transactions Component
+
+Here is a class diagram of the `Transactions` component
+![TransactionsClassDiagram](images/TransactionsClassDiagram.png)
+
+The `Transactions` component
+
+- A `Transaction` is recorded upon every command that changes the balance of an account
+- Certain commands allow the provision of a description parameter to associate a custom description to a transaction
+- `Transactions` are only accessed through the `TransactionManager`
+- `Transactions` associated to an account are deleted if and only if the account is deleted
+
+## Implementation
+
+### Create/Delete account feature
+
+This feature is facilitated by `AccountList` Class within the `Accounts` Component.
+The method called from `AccountList` is the `addAccount` method which creates a new `Account` object.The `deleteAccount`
+method called from `AccountList` delete the specified `Account` object.
+
+The current implementation initialises the `Account` with 0 balance.
+Only currency account which 0 balance can be deleted.
+
+Given below is an example of the usage of this feature and the mechanism at each step
+
+Step 1: The user launches the application for the first time and `AccountList` is created with no `Account`
+![AccountListObjectDiagram1](images/AccountListObjectDiagram1.png)
+
+Step 2: The user passes in the command `create-account <CURRENCY>`, where `CURRENCY` is a valid string representing one
+of the elements of the `Currency` enum
+![AccountListObjectDiagram1](images/AccountListObjectDiagram2.png)
+
+Step 3: The user passes in the command `create-account <CURRENCY>`, where `CURRENCY` is also valid but different to that
+in step 1.
+![AccountListObjectDiagram1](images/AccountListObjectDiagram3.png)
+
+Step 4: The user passes the command `delete-account CURRENCY`, for example `delete-account SGD`.
+
+![AccountListObjectDiagram4](images/AccountListObjectDiagram4.png)
+
+The following sequence diagram shows how the Create Account operation works
```

Comment:
Maybe it is better to align the starting point of `execute` at the start of the program and the ending point of final return dashed line at the end of the program  to be on the same vertical line.
![image](https://user-images.githubusercontent.com/16986946/228762947-ad45a97c-21e9-4a3c-bd5d-bb556a5d2ceb.png)